python/examples/integrations/bentoml/README.md

```diff
@@ -0,0 +1,166 @@
+
+There are two notable files in this example:
+
+- `train.py` - Trains and saves a versioned kNN model with scikit-learn, BentoML, and the iris dataset.
+- `service.py` - Creates a model prediction API endpoint with BentoML and writes whylogs data profiles to WhyLabs for ML monitoring.
+
+## 🛠️ How to implement ML monitoring for BentoML with whylogs and WhyLabs:
+
+After running the training script with `python train.py`, you'll see a message similar to "`Model saved: Model(tag="iris_knn:fhfsvifgrsrtjlg6")`". This is the versioned model saved to the BentoML local model store.
+
+In `service.py`, we'll get ready to serve and monitor our latest saved model.
+
+To get the WhyLabs API key and org-id, go to the [WhyLabs dashboard](https://hub.whylabsapp.com/) and click on the "Access Tokens" section in the settings menu. Model-ids are found under the "Project Management" tab in the same section.
+
+```python
+import os
+
+import numpy as np
+import whylogs as why
+
+import bentoml
+from bentoml.io import NumpyNdarray, Text
+
+# Set WhyLabs environment variables
+os.environ['WHYLABS_API_KEY'] = 'APIKEY'
+os.environ["WHYLABS_DEFAULT_ORG_ID"] = 'ORGID'
+os.environ["WHYLABS_DEFAULT_DATASET_ID"] = 'PROJECTID'
+
+# Model Prediction Class Names
+CLASS_NAMES = ["setosa", "versicolor", "virginica"]
+
+# Load model from BentoML local model store & create a BentoML runner
+iris_clf_runner = bentoml.sklearn.get("iris_knn:latest").to_runner()
+svc = bentoml.Service("iris_classifier", runners=[iris_clf_runner])
+
+```
+
+> Pro tip: A best practice for software deployment is to have these environment variables always set on the machine/environment level (such as per the CI/QA machine, a Kubernetes Pod, etc.) so it is possible to leave code untouched and persist it correctly according to the environment you execute it.
+
+Now that our bentoML runner and environment variables are set, we can create an API endpoint with BentoML and set up model and data monitoring with whylogs and WhyLabs. We'll Log the data, predicted class, and model probability score.
+
+```python
+# Create a BentoML API endpoint for model predictions & logging
+@svc.api(
+    input=NumpyNdarray.from_sample(np.array([4.9, 3.0, 1.4, 0.2],
+    dtype=np.double)),
+    output=Text(),
+)
+async def classify(features: np.ndarray) -> str:
+    results = await iris_clf_runner.predict.async_run([features])
+    probs = iris_clf_runner.predict_proba.run([features])
+    result = results[0]
+
+    category = CLASS_NAMES[result]
+    prob = max(probs[0])
+
+    # create a dict of data & prediction results w/ feature names
+    data = {
+        "sepal length": features[0],
+        "sepal width": features[1],
+        "petal length": features[2],
+        "petal width": features[3],
+        "class_output": category,
+        "proba_output": prob,
+    }
+
+    # Log data+outputs with whylogs & write to WhyLabs.ai
+    profile_results = why.log(data)
+    profile_results.writer("whylabs").write()
```

Comment:
ok, wondering if this is the best approach to take with bento, since it will be a live service. wouldn't it be best to have a pre-configured rolling logger?
wondering because it can be hard on WhyLabs' side to handle several (order of 1000s) one-line profile writes, which would come on every `classify` call

Reply:
Good point. I am having trouble finding how to easily call a startup/shutdown function like that to initialize the rolling logger. I'll ask about in their community.

Reply:
They don't have lifecycle built in right now, but I could probably init the roller in service.py and use it in the prediction function.

Reply:
are we solved on this?

Reply:
Resolved: startup and shutdown logic implemented with a class